src/ClimaCoupler.jl

```diff
@@ -5,6 +5,18 @@ Coupling module sufficient for initial atmos-ocean-land coupled simulation.
 """
 module ClimaCoupler
 
+using ClimaCore
+import ClimaCore: Fields, Operators
+import Logging
+
+# Disable errors from broadcasting with mismatched spaces
+Fields.allow_mismatched_diagonalized_spaces() = true
+Operators.allow_mismatched_fd_spaces() = true
```

Comment:
This should be moved into the test folder, it must be done at global scope.

Reply:
The rest of the changes in this file can probably be deleted

Reply:
Can you explain your reasoning for both of these points? I was thinking we want to disable these errors anytime we are using the coupler. Maybe that is simply too broad? The benefit is that coupler users won't have to think about these levers.

Reply:
This is kind of like type piracy. We should probably avoid changing other packages default behavior under the hood. What's the issue with doing this at the global level? I'm fairly certain it'll work, it's simple, and more transparent.

Reply:
That makes sense to me r.e. transparency. I just wanted to allow the user to not have to think about this when coupling and these errors are something we will want to disable pretty much every time we are coupling, therefore something more automatic seemed warranted. 

Maybe a method that redefines these would be a happy medium? Something like
```julia
function allow_mismatched_spaces()
    @eval Fields.allow_mismatched_diagonalized_spaces() = true
    @eval Operators.allow_mismatched_fd_spaces() = true
end
```

I just feel that `allow_mismatched_diagonalized_spaces` and `allow_mismatched_fd_spaces` are pretty niche.

Reply:
> I just wanted to allow the user to not have to think about this

This is not transparent.

My recommendation is to include this at the top level. I recommend not using `@eval`.

Reply:
> I recommend not using @eval. My recommendation is to include this at the top level. That is by far the simplest way to handle this situation. We can hide it later if we want to make things pretty.

Ok, I'm fine with holding off then.